includes/admin/class-wc-rest-payments-survey-controller.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * Class WC_REST_Payments_Survey_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for settings.
+ */
+class WC_REST_Payments_Survey_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/upe_survey';
+
+	/**
+	 * The HTTP client, used to forward the request to WPCom.
+	 *
+	 * @var WC_Payments_Http
+	 */
+	protected $http_client;
+
+	/**
+	 * The constructor.
+	 * WC_REST_Payments_Survey_Controller constructor.
+	 *
+	 * @param WC_Payments_Http_Interface $http_client - The HTTP client, used to forward the request to WPCom.
+	 */
+	public function __construct( WC_Payments_Http_Interface $http_client ) {
+		$this->http_client = $http_client;
+	}
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'submit_survey' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+				'args'                => [
+					'reasons'  => [
+						'type'              => 'array',
+						'items'             => [
+							'type' => 'string',
+							'enum' => [
+								'slow-buggy',
+								'theme-compatibility',
+								'missing-features',
+								'store-sales',
+								'poor-customer-experience',
+							],
+						],
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+					'comments' => [
+						'type'              => 'string',
+						'validate_callback' => 'rest_validate_request_arg',
```

Comment:
Should we add a `sanitize_callback` for comments since it's freeform text?

![Screen Shot 2021-07-08 at 3 28 07 PM](https://user-images.githubusercontent.com/6723003/124930145-5cb6b900-e001-11eb-94ae-04ccf18f77a5.png)

Reply:
The tools seem to already ensure that the text is not executing JS, but I guess adding a `wp_filter_nohtml_kses` won't hurt - done in 4cb234cc

It still won't prevent somebody from posting HTML directly to the endpoint, tho 😄

Reply:
👍